aspect-build/rules_js
# feat: Support hermetic patching

Allows Bazel native patch to be used with npm_import(). This removes the existing dependency on bash when enabled. The proviso is that patch diff headers must have an unstripped 'package/' prefix when use_native_patch is true.

### Type of change

- New feature or functionality (change which adds functionality)
- Documentation (updates to documentation or READMEs)

**For changes visible to end-users**

Suggested release notes are provided below:

- use_native_patch flag added to npm_import, allowing for hermetic patching.

### Test plan

Manual testing steps:

* set BAZEL_SH var to a nonexistent path.
* Create valid patch. Patch diff header should contain package prefix, e.g. "diff --git package/lib/font.coffee ..."
* verify that npm_import() with patch applied fails "Error applying patch". System bash cannot be found.
* verify that npm_import() with patch applied and use_native_patch set to 'True' succeeds.
